src/controller/cna.controller/cna.controller.js

```diff
@@ -35,231 +35,249 @@ async function getCna (req, res) {
+              logger.error(err.stack)
+              return res.status(500).send('Internal Server Error')
+            }
+
+            const responseMessage = {
+              message: newCna.short_name + ' CNA was successfully created.',
+              created: newCna
+            }
+
+            logger.info(JSON.stringify(responseMessage))
+            return res.status(200).json(responseMessage)
+          })
       } else {
-        logger.info(cna.name + ' CNA was successfully updated in MongoDB')
+        logger.info(newCna.short_name + ' CNA was not created because it already exists.')
+        return res.status(400).json({ message: newCna.short_name + ' CNA already exists.' })
       }
-
-      const responseMessage = {
-        message: shortName + ' CNA was successfully created',
-        created: cna
-      }
-
-      return res.status(200).json(responseMessage)
     })
 }
 
 // TODO: Request body sanitation and parameter validation
+// Updates a cna only if the cna exist for the specified shortname. If no cna exists, it does not create the cna.
 async function updateCna (req, res) {
-  const shortName = req.params.id
-  const cna = {}
+  const shortName = req.params.shortname
+  const cna = req.body
 
   if (req.body.short_name !== undefined) {
-    if (typeof req.body.short_name !== 'string') {
-      return res.status(400).json({ message: 'Wrong value type' })
-    }
-
     cna.short_name = req.body.short_name
+  } else {
+    cna.short_name = shortName
   }
 
-  if (req.body.name !== undefined) {
-    if (typeof req.body.name !== 'string') {
-      return res.status(400).json({ message: 'Wrong value type' })
-    }
-
-    cna.name = req.body.name
-  }
-
+  const newCna = new Cna(cna)
   Cna.findOneAndUpdate()
-    .byShortName(shortName)
-    .updateOne(cna)
+    .byShortName(newCna.short_name)
+    .updateOne(newCna)
     .exec((err, result) => {
       if (err) {
         logger.error(err.stack)
         return res.status(500).send('Internal Server Error')
       }
 
       if (result.n === 0) {
-        logger.info(shortName + ' CNA was not updated in MongoDB because it does not exists in the database.')
-        return res.status(200).json({ message: 'The cna does not exist' })
-      } else {
-        logger.info(shortName + ' CNA was successfully updated in MongoDB')
+        logger.info(newCna.short_name + ' CNA could not be updated in MongoDB because it does not exist.')
+        return res.status(400).json({ message: newCna.short_name + ' CNA could not be updated in MongoDB because it does not exist.' })
       }
 
       const responseMessage = {
-        message: 'The cna was successfully updated',
-        updated: cna
+        message: 'The cna was successfully updated in MongoDB.',
+        updated: newCna
       }
 
+      logger.info('The cna was successfully updated in MongoDB.')
       return res.status(200).json(responseMessage)
     })
 }
 
 // TODO: Request body sanitation and parameter validation
```

Comment:
Should this be removed to maintain consistency throughout the controller?

Reply:
So something has actually gone horribly wrong. I rebased on dev to get updates for the contributing guide and somehow that has resulted in code changes. Some commits must have snuck into dev? I'm going to have to see if I can remove that commit and fix it.

Reply:
Commits from the pre-release to dev?